lib/msf/core/exploit/retry.rb

```diff
@@ -0,0 +1,33 @@
+module Msf::Exploit::Retry
+  # Retry the block until it returns a truthy value. Each iteration attempt will
+  # be performed with an exponential backoff. If the timeout period surpasses,
+  # false is returned.
+  #
+  # @param Integer timeout the number of seconds to wait before the operation times out
+  def retry_until_true(timeout:)
```

Comment:
Not a blocker:

```suggestion
  def retry_until_truthy(timeout:)
```

I'm not sure why past me preferenced `true` over `truthy`, when I called out `truthy` explicitly in the yard docs. Not a blocker for me though.